aten/src/ATen/native/cuda/CuFFTPlanCache.h

```diff
@@ -69,6 +120,94 @@ static bool is_pow_of_two(int64_t x) {
   return (x & (x - 1)) == 0;
 }
 
+#ifdef __HIP_PLATFORM_HCC__
+    using cufft_size_type = int;
+#else
+    using cufft_size_type = long long int;
+#endif
+
+using CuFFTDimVector = c10::SmallVector<cufft_size_type, at::kDimVectorStaticSize>;
+struct CuFFTDataLayout {
+  CuFFTDimVector embed;
+  cufft_size_type stride, dist;
+  bool must_clone, simple;
+};
+
+// Returns a simple (contiguous) cufft embedding for a signal of the given size
```

Comment:
Elaborate on how this is intended to be used. Or, alternatively, why it's necessary